### PR TITLE
PAYARA-3676 PAYARA-3665 Fix More Asadmin Prompt Issues

### LINKED ISSUE #3833: Creating a domain with admin password results in values do not match and failure PAYARA-3676
# Description #
----------

This is for Payara Server 5.191. Doesn't happen in 5.184. Linux OS (Ubuntu Bionic LTS)

When creating a new domain with `asadmin create-domain` and typing admin password, the command prints "Enter the admin password" again after the first key press. When asked to enter the password again, reports that the passwords don't match even if they are the same.

## Expected Outcome

* The `create-domain` command executes successfully
* A new domain `newdomain` is created
* the new domain is secured by admin user with password `password`
* running `bin/asadmin list-applications` asks for password, when `password` entered, executes successfully

## Current Outcome

* the command prompts for admin password
* after first key press, the command prints the prompt message again
* after entering full password, the command asks for entering the password again
* the same password is entered again
* the command complains that "CLI173: Values entered for Enter the admin password do not match."
* the command results in a failure

This is a log from the command executed in a terminal:

```
> bin/asadmin create-domain newdomain     
Enter admin user name [Enter to accept default "admin" / no password]>Enter the admin password [Enter to accept default of no password]> 
Enter the admin password again> 
CLI173: Values entered for Enter the admin password do not match.
Usage: asadmin [asadmin-utility-options] create-domain
	[--adminport <adminport(default:4848)>] [--portbase <portbase>]
	[--template <template>] [--domaindir <domaindir>]
	[--instanceport <instanceport(default:8080)>]
	[--savemasterpassword[=<savemasterpassword(default:false)>]]
	[--usemasterpassword[=<usemasterpassword(default:false)>]]
	[--domainproperties <domainproperties>]
	[--keytooloptions <keytooloptions>]
	[--savelogin[=<savelogin(default:false)>]]
	[--nopassword[=<nopassword(default:false)>]]
	[--checkports[=<checkports(default:true)>]]
	[-?|--help[=<help(default:false)>]] domain_name
Command create-domain failed.
```

## Steps to reproduce (Only for bug reports) 

* run `bin/asadmin create-domain newdomain`
* when asked for admin password, type password `password`
* when asked to re-enter the password again, type password `password` again

### A workaround

A workaround is to press any key before entering the password again. Pressing any key will trigger printing the prompt message again. What is typed from this point is taken as a password.

In other words, if the password is "password", then type "xpassword" when first asked for a password (the command will print the prompt message after pressing x). Then enter password "password" when asked to enter the password again.

## Context (Optional)

This happens when creating a new domain interactively and entering admin password when prompted. It's probably caused by an upgrade to JLink.

## Environment ##

- **Payara Version**: 5.191
- **Edition**: Full
- **JDK Version**: OpenJDK 8
- **Operating System**: Linux (Ubuntu Bionic)

### PR DESCRIPTION
**Pull request by Pandrex247**
Much more aggressive use of JLine to hopefully catch any remaining issues around asadmin prompts exiting after a single input character.

Fixes #3833

### CODE REVIEW

File: nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java [hunk 1]
```diff
@@ -120,21 +121,32 @@ protected int executeCommand() throws CommandException, CommandValidationExcepti
...
-        }
-        if (cons != null) {
-            cons.printf("%s", strings.get("AdminUserPrompt", defuser));
-            String val = cons.readLine();
-            if (val != null && val.length() > 0){
-                user = val;
-            } else {
-                user = defuser;
+
+        try (ConsoleReader cons = new ConsoleReader(System.in, System.out, null)) {
```
Comment (jbee) [inline nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java]: `cons` should be `console`

File: nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java [hunk 2]
```diff
@@ -120,21 +121,32 @@ protected int executeCommand() throws CommandException, CommandValidationExcepti
...
+                    logger.log(Level.WARNING, "Error reading input", ioe);
+                }
+                if (val != null && val.length() > 0){
+                    user = val;
+                } else {
+                    user = defuser;
+                }
+            }
+        } catch (IOException ioe) {
+            logger.log(Level.WARNING, "Error reading input", ioe);
```
Comment (jbee) [inline nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java]: There is already a try-catch wrapping `cons.readLine()`. Is this one necessary?

File: nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/remote/RemoteCLICommand.java [hunk 3]
```diff
@@ -171,52 +173,62 @@ public void fetchCommandModel() throws CommandException {
          */
         @Override
         protected boolean updateAuthentication() {
```
Comment (jbee) [inline nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/remote/RemoteCLICommand.java]: Looking at the changes of this file and the one before there seems to be a common structure for the commands that maybe can be extracted so the code does not need to be "duplicated"?

File: nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java (hunk 1 above)
Comment (Pandrex247) [reply nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java]: I was just keeping it the same, but agreed I'll change it

File: nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java (hunk 2 above)
Comment (Pandrex247) [reply nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java]: That was a refactor I missed, no it's not necessary as the log messages are the same - I originally had them different (_error instantiating_ and _error reading_).

File: nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/remote/RemoteCLICommand.java (hunk 3 above)
Comment (Pandrex247) [reply nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/remote/RemoteCLICommand.java]: sUt#44f6uUn3

Comment (Pandrex247) [reply nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/remote/RemoteCLICommand.java]: I saw that but didn't include it as I'm not a fan of bringing refactors into bug fixes.

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Pandrex247** [comment]
Jenkins test please

**Pull request merged**